types/index.d.ts

```diff
@@ -98,7 +98,7 @@ export interface MockResponseInit extends MockParams {
 export type ErrorOrFunction = Error | ((...args: any[]) => Promise<any>);
 export type UrlOrPredicate = string | RegExp | ((input: Request) => boolean);
 
-export type MockResponseInitFunction = (request: Request) => MockResponseInit | string | Promise<MockResponseInit | string>;
+export type MockResponseInitFunction = (request: Request) => MockResponseInit | string | Promise<MockResponseInit | string | Response>;
```

Comment:
For symmetry, `Response` should also be acceptable without `Promise`.
```suggestion
export type MockResponseInitFunction = (request: Request) => MockResponseInit | string | Response | Promise<MockResponseInit | string | Response>;
```